MATLAB/moordyn_sfunc_3dof.cpp

```diff
@@ -195,79 +196,78 @@ static void mdlInitializeSizes(SimStruct *S)
+ *    start of simulation and if it is present in an enabled subsystem
+ *    configured to reset states, it will be call when the enabled subsystem
+ *    restarts execution to reset the states.
+ */
+static void mdlInitializeConditions(SimStruct* S)
+{
+}
 #endif /* MDL_INITIALIZE_CONDITIONS */
 
 
 
 #define MDL_START  /* Change to #undef to remove function */
 #if defined(MDL_START) 
-  /* Function: mdlStart =======================================================
-   * Abstract:
-   *    This function is called once at start of model execution. If you
-   *    have states that should be initialized once, this is the place
-   *    to do it.
-   */
-  static void mdlStart(SimStruct *S)
-  {
-    double *InputAry = (double *)ssGetDWork(S, WORKARY_INPUT); //malloc(NumInputs*sizeof(double));   
-    double *OutputAry = (double *)ssGetDWork(S, WORKARY_OUTPUT);
-    if (t == -1){
+/* Function: mdlStart =======================================================
+ * Abstract:
+ *    This function is called once at start of model execution. If you
+ *    have states that should be initialized once, this is the place
+ *    to do it.
+ */
+static void mdlStart(SimStruct* S)
+{
+    double* InputAry = (double*)ssGetDWork(S, WORKARY_INPUT); //malloc(NumInputs*sizeof(double));   
+    double* OutputAry = (double*)ssGetDWork(S, WORKARY_OUTPUT);
+    if (t == -1) {
         
         // get general system definitions
-        ndof = mdSystem->NCoupledDOF();
+        //ndof = mdSystem->NCoupledDOF();
         auto points = mdSystem->GetPoints();
         auto lines = mdSystem->GetLines();
         int num_lines = lines.size();
         double x[ndof], xd[ndof];
-
+        
         // get initial points
-        for (size_t i=0; i < num_lines; i++) {
-            auto init_pts = points[i*2]->getPosition(); // even #s are partition pts, odd #s are anchors
-            x[i*3] = static_cast<double>(init_pts[0]);
-            x[i*3+1] = static_cast<double>(init_pts[1]);
-            x[i*3+2] = static_cast<double>(init_pts[2]);
+        for (size_t i = 0; i < num_lines; i++) {
+            auto init_pts = points[i * 2]->getPosition(); // even #s are partition pts, odd #s are anchors
+            x[i * 3] = static_cast<double>(init_pts[0]);
+            x[i * 3 + 1] = static_cast<double>(init_pts[1]);
+            x[i * 3 + 2] = static_cast<double>(init_pts[2]);
         }
-         // initial velocity is zero
+        // initial velocity is zero
         memset(xd, 0.0, sizeof(double));
         
         // Setup for time series simulation
         int err = mdSystem->Init(x, xd);
         if (checkError(S)) return;
-        t = 0;
-
+        t = 0.0;
+        
         // Calculate forces at t=0
-        err = mdSystem->Step(x, xd, OutputAry, t, dt);
-
-        if (checkError(S)) return;
-        setOutputs(S, OutputAry);
+        //err = mdSystem->Step(x, xd, OutputAry, t, dt);
+        //if (checkError(S)) return;
+        //setOutputs(S, OutputAry);
```

Comment:
Is there a reason this was commented out? I added this code in because it was what was causing MoorDyn to return 0 at t=0, which was breaking OpenFresco at the time.